lib/Service/U2FManager.php

```diff
@@ -69,6 +72,7 @@ public function disableU2F(IUser $user) {
 		// TODO: use single query instead
 		foreach ($this->mapper->findRegistrations($user) as $registration) {
 			$this->mapper->delete($registration);
+			$this->publishEvent($user, 'u2f_device_removed');
```

Comment:
Can an admin trigger this? Then you should also `setAuthor` and translate it differently.